### PR TITLE
Install pip via apt

### PR DESCRIPTION
**Pull request by Eder-K**
Closes #97.

This changes the `pip` installation method: we now use `pip` provided by `apt`.

Note: the installation of `pip` should occur in the dockerfile that requires it, not in earlier base files. At the moment, only `fenics-adapter` requires `pip` to function, so it is the only adapter affected by this change.

In Ubuntu 18.04, we have found that the `pip` version provided by `apt` is too old, therefore we add the following step after installing `pip`:
```
pip3 install --user --upgrade pip
```

For context, the discussion below also mentions additional, unrelated changes to the dockerfiles that were initially attached to this PR. These changes have since moved to https://github.com/precice/systemtests/pull/269.

### CODE REVIEW

File: adapters/Dockerfile.fenics-adapter [hunk 1]
```diff
@@ -18,32 +23,28 @@ RUN apt-get -qq update && apt-get -qq install \
     python3-dev && \
     rm -rf /var/lib/apt/lists/*
 
-# Install pip dependecies; we have to use pip, since cython provided by apt-get is too old.
-RUN wget -q https://bootstrap.pypa.io/get-pip.py -O get-pip.py && \
-    python3 get-pip.py
-
-
+# Install fenics
```
Comment (BenjaminRodenberg) [inline adapters/Dockerfile.fenics-adapter]:
```suggestion
# Install FEniCS
```

File: adapters/Dockerfile.su2-adapter [hunk 2]
```diff
@@ -1,10 +1,14 @@
+# Dockerfile for building the su2-adapter for preCICE
+# NOTE: Built for SU2 v6.0.0
+
 ARG branch=develop
-ARG from=precice/precice-ubuntu1604.home-${branch}
+ARG from=precice/precice-ubuntu1804.home-${branch}
```
Comment (BenjaminRodenberg) [inline adapters/Dockerfile.su2-adapter]: I'm not sure whether this kind of change should really go into this PR.

File: precice/Dockerfile.Ubuntu1604.home [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Dockerfile for building preCICE on ubuntu 16.04
 
+# #################
+# WARNING: As of 11.2020, images using Ubuntu 16.04 are no longer maintained.
+#          Attempting to build this dockerfile or using it as base image for
+#          other builds is likely to fail.
+# #################
```
Comment (BenjaminRodenberg) [inline precice/Dockerfile.Ubuntu1604.home]: See above. Not really sure whether this should go into this PR.

File: precice/Dockerfile.Ubuntu1804.home [hunk 4]
```diff
@@ -14,63 +21,63 @@ RUN apt-get -qq update && apt-get -qq install \
...
+cd petsc-3.12.5 && \
+./configure && \
+make all test
 RUN chown -R precice:precice /petsc
 
-# Rebuild image if force_rebuild after that command
+USER precice
+
+# Upgrade pip to latest version
+RUN pip3 install --user --upgrade pip
```
Comment (BenjaminRodenberg) [inline precice/Dockerfile.Ubuntu1804.home]: If we already provide the latest version of pip with this image, why do we install pip (again?) [here](https://github.com/precice/systemtests/pull/260/files?file-filters%5B%5D=.calculix-adapter&file-filters%5B%5D=.codeaster-adapter&file-filters%5B%5D=.dealii-adapter_3D&file-filters%5B%5D=.fenics-adapter&file-filters%5B%5D=.home&file-filters%5B%5D=.mpich&file-filters%5B%5D=.openfoam-adapter&file-filters%5B%5D=.package&file-filters%5B%5D=.su2-adapter&file-filters%5B%5D=.yml#diff-3a5025fdfc7d5ffbc4d1140e2ef5014804df9f0e376db4c02177e3933de08b79L43)?

Comment (BenjaminRodenberg) [reply precice/Dockerfile.Ubuntu1804.home]: If I understand it correctly `python3-pip` is no preCICE dependency. Therefore, I would suggest to **not** install it in `precice/Dockerfile.Ubuntu1234.xyz`, but where we actually need pip the first time (bindings, fenics-adapter, code-aster). This allows us to make sure that preCICE does not (by accident) depend on pip.

**Review by BenjaminRodenberg — CHANGES_REQUESTED** [review]
Using pip from apt is a good step. However, I have a few minor comments below.

File: precice/Dockerfile.Ubuntu1604.home (hunk 3 above)
Comment (Eder-K) [reply precice/Dockerfile.Ubuntu1604.home]: Fair point, I moved the comment cleanup related changes to a new PR (see https://github.com/precice/systemtests/pull/269), this PR now only handles the apt pip change. 

I force pushed my local branch here to keep the commit mess to a minimum, sorry about that :sweat_smile: At least the `git add -p` function proved to be very useful once more.

File: precice/Dockerfile.Ubuntu1804.home (hunk 4 above)
Comment (Eder-K) [reply precice/Dockerfile.Ubuntu1804.home]: I agree. In fact, it seems that `fenics-adapter` is currently the only adapter in need of pip. I removed the pip installs from the remaining precice dockerfiles and added the pip upgrade step to `fenics-adapter`.
It should be noted that if an adapter/test does require pip in the future, one should add `pip3 install --user --upgrade pip` as mentioned above to make sure that the version is sufficient at all times (currently needed for Ubuntu 18.04).

File: adapters/Dockerfile.fenics-adapter (hunk 1 above)
Comment (Eder-K) [reply adapters/Dockerfile.fenics-adapter]: Added.

File: adapters/Dockerfile.su2-adapter (hunk 2 above)
Comment (Eder-K) [reply adapters/Dockerfile.su2-adapter]: See comments below.

**Review by BenjaminRodenberg — APPROVED** [review]
Looks good :+1:

**Pull request merged**